helix-term/src/commands.rs

```diff
@@ -2543,9 +2551,10 @@ fn buffer_picker(cx: &mut Context) {
                 .path
                 .as_deref()
                 .map(helix_core::path::get_relative_path);
-            let path = match path.as_deref().and_then(Path::to_str) {
-                Some(path) => path,
-                None => SCRATCH_BUFFER_NAME,
```

Comment:
The existing code here was actually incorrectly returing `SCRATCH_BUFFER_NAME` when the `Path::to_str` call failed to validate the name as UTF-8. It should have been using `Path::to_string_lossy`, like:

```rust
let path = path
    .as_deref()
    .map_or_else(|| SCRATCH_BUFFER_NAME.into(), Path::to_string_lossy);
```

I think we should make a similar fix here after handling the Result from `get_relative_path`. But if `get_relative_path` fails then we should use the full absolute path rather than the `SCRATCH_BUFFER_NAME`.

Same thing for the jumplist picker below.

Reply:
Done !
 
I have done the same on the jumplist_picker just below